pkg/kubectl/cmd/get/get_test.go

```diff
@@ -1301,18 +1303,205 @@ func TestWatchResource(t *testing.T) {
+`,
+		},
+		{
+			format: "json",
+			expected: `{"type":"ADDED","object":{"apiVersion":"v1","kind":"Pod","metadata":{"creationTimestamp":null,"name":"bar","namespace":"test","resourceVersion":"9"},"spec":{"containers":null,"dnsPolicy":"ClusterFirst","enableServiceLinks":true,"restartPolicy":"Always","securityContext":{},"terminationGracePeriodSeconds":30},"status":{}}}
+{"type":"ADDED","object":{"apiVersion":"v1","kind":"Pod","metadata":{"creationTimestamp":null,"name":"foo","namespace":"test","resourceVersion":"10"},"spec":{"containers":null,"dnsPolicy":"ClusterFirst","enableServiceLinks":true,"restartPolicy":"Always","securityContext":{},"terminationGracePeriodSeconds":30},"status":{}}}
+{"type":"MODIFIED","object":{"apiVersion":"v1","kind":"Pod","metadata":{"creationTimestamp":null,"name":"foo","namespace":"test","resourceVersion":"11"},"spec":{"containers":null,"dnsPolicy":"ClusterFirst","enableServiceLinks":true,"restartPolicy":"Always","securityContext":{},"terminationGracePeriodSeconds":30},"status":{}}}
+{"type":"DELETED","object":{"apiVersion":"v1","kind":"Pod","metadata":{"creationTimestamp":null,"name":"foo","namespace":"test","resourceVersion":"12"},"spec":{"containers":null,"dnsPolicy":"ClusterFirst","enableServiceLinks":true,"restartPolicy":"Always","securityContext":{},"terminationGracePeriodSeconds":30},"status":{}}}
+`,
+		},
+		{
+			format: "yaml",
+			expected: `object:
+  apiVersion: v1
+  kind: Pod
+  metadata:
+    creationTimestamp: null
+    name: bar
+    namespace: test
+    resourceVersion: "9"
+  spec:
+    containers: null
+    dnsPolicy: ClusterFirst
+    enableServiceLinks: true
+    restartPolicy: Always
+    securityContext: {}
+    terminationGracePeriodSeconds: 30
+  status: {}
+type: ADDED
+
+object:
+  apiVersion: v1
+  kind: Pod
+  metadata:
+    creationTimestamp: null
+    name: foo
+    namespace: test
+    resourceVersion: "10"
+  spec:
+    containers: null
+    dnsPolicy: ClusterFirst
+    enableServiceLinks: true
+    restartPolicy: Always
+    securityContext: {}
+    terminationGracePeriodSeconds: 30
+  status: {}
+type: ADDED
+
+object:
+  apiVersion: v1
+  kind: Pod
+  metadata:
+    creationTimestamp: null
+    name: foo
+    namespace: test
+    resourceVersion: "11"
+  spec:
+    containers: null
+    dnsPolicy: ClusterFirst
+    enableServiceLinks: true
+    restartPolicy: Always
+    securityContext: {}
+    terminationGracePeriodSeconds: 30
+  status: {}
+type: MODIFIED
+
+object:
+  apiVersion: v1
+  kind: Pod
+  metadata:
+    creationTimestamp: null
+    name: foo
+    namespace: test
+    resourceVersion: "12"
+  spec:
+    containers: null
+    dnsPolicy: ClusterFirst
+    enableServiceLinks: true
+    restartPolicy: Always
+    securityContext: {}
+    terminationGracePeriodSeconds: 30
+  status: {}
+type: DELETED
+
+`,
+		},
+		{
+			format: `jsonpath={.type},{.object.metadata.name},{.object.metadata.resourceVersion}{"\n"}`,
```

Comment:
this is nifty... allows for `oc observe`-like invocations

Reply:
yeah, good